consensus/core/src/commit_syncer.rs

```diff
@@ -430,19 +439,42 @@ impl<C: NetworkClient> CommitSyncer<C> {
             fetched_blocks.extend(result?);
         }
 
+        // 8. Make sure fetched block timestamps are lower than current time.
+        for block in &fetched_blocks {
+            let now_ms = timestamp_utc_ms();
+            let forward_drift = block.timestamp_ms().saturating_sub(now_ms);
+            if forward_drift == 0 {
+                continue;
+            };
+            let peer_hostname = &inner.context.committee.authority(target_authority).hostname;
+            inner
+                .context
+                .metrics
+                .node_metrics
+                .block_timestamp_drift_wait_ms
+                .with_label_values(&[peer_hostname, &"commit_syncer"])
+                .inc_by(forward_drift);
+            sleep(Duration::from_millis(forward_drift)).await;
```

Comment:
we need an additional check so that we are not waiting for too long here, maybe use  `context.parameters.max_forward_time_drift`?